### PR TITLE
pre-commit autoupdate 2024-08-16

### PR DESCRIPTION
**Pull request by cclauss**
## Description

Update pre-commit dependencies.  These pull requests can be automated via https://pre-commit.ci

## Developer Certificate of Origin
We need all contributions to comply with the [Developer Certificate of Origin (DCO)](https://developercertificate.org/). All commits must be signed off by including a `Signed-off-by` line in the commit message. [See our wiki for guidance](https://github.com/kedro-org/kedro/wiki/Guidelines-for-contributing-developers/).

If your PR is blocked due to unsigned commits, then you must follow the instructions under "Rebase the branch" on the GitHub Checks page for your PR. This will retroactively add the sign-off to all unsigned commits and allow the DCO check to pass.

## Checklist

- [x] Read the [contributing](https://github.com/kedro-org/kedro/blob/main/CONTRIBUTING.md) guidelines
- [x] Signed off each commit with a [Developer Certificate of Origin (DCO)](https://docs.github.com/en/repositories/managing-your-repositorys-settings-and-features/managing-repository-settings/managing-the-commit-signoff-policy-for-your-repository)
- [ ] Opened this PR as a 'Draft Pull Request' if it is work-in-progress
- [ ] Updated the documentation to reflect the code changes
- [ ] Added a description of this change in the [`RELEASE.md`](https://github.com/kedro-org/kedro/blob/main/RELEASE.md) file
- [x] Added tests to cover my changes
- [ ] Checked if this change will affect Kedro-Viz, and if so, communicated that with the Viz team

### CODE REVIEW

**Comment by astrojuanlu** [comment]
Thanks a lot @cclauss!

@kedro-org/kedro-framework Should we make these pre-commit updates automatic? Also please review

**Review by DimedS — APPROVED** [review]
Thanks, @cclauss and @astrojuanlu! I think it’s a great idea to create these PRs on a regular basis, maybe every two weeks or once a month, if there have been any changes during that period.

**Comment by cclauss** [comment]
Manual or automated?

From the commit message:
> These pull requests can be automated via https://pre-commit.ci/

File: docs/source/notebooks_and_ipython/notebook-example/add_kedro_to_a_notebook.ipynb
[the remark this answers is not in the document]
Comment (cclauss) [reply docs/source/notebooks_and_ipython/notebook-example/add_kedro_to_a_notebook.ipynb:1]: Let's not encourage learners to swim upstream against [PEP8](https://peps.python.org/pep-0008/#imports).
> Imports are always put at the top of the file, just after any module comments and docstrings, and before module globals and constants.

[the remark this answers is not in the document]
Comment (astrojuanlu) [reply docs/source/notebooks_and_ipython/notebook-example/add_kedro_to_a_notebook.ipynb:1]: I don't think there was a particular intent behind the order of these `import` statements, I'm fine with the new arrangement made by ruff 0.6+ 👍🏼

[the remark this answers is not in the document]
Comment (merelcht) [reply docs/source/notebooks_and_ipython/notebook-example/add_kedro_to_a_notebook.ipynb:1]: Thanks for confirming @astrojuanlu! In that case, ignore my comment 😄

**Review by merelcht — APPROVED** [review]
Thanks for the contribution @cclauss ⭐ !

**Pull request merged**